uint32 red
uint32 green
